bson/unmarshaling_cases_test.go

```diff
@@ -136,6 +154,20 @@ func unmarshalingTestCases() []unmarshalingTestCase {
 				Foo: []byte{0, 1, 2, 3, 4, 5},
 			}),
 		},
+		{
```

Comment:
Optional: Consider adding comments that reference the ticket number (GODRIVER-2427) and a brief explanation of what each new test case intends to test. See above test cases for examples.